snuba/admin/clickhouse/system_queries.py

```diff
@@ -129,13 +112,60 @@ def _run_sql_query_on_host(
 )
 
 
-def is_query_select(sql_query: str) -> bool:
+def is_query_using_only_system_tables(
+    clickhouse_host: str,
+    clickhouse_port: int,
+    storage_name: str,
+    sql_query: str,
+) -> bool:
     """
-    Simple validation to ensure query is a select command
+    Run the EXPLAIN QUERY TREE on the given sql_query and check that the only tables
+    in the query are system tables.
     """
-    sql_query = " ".join(sql_query.split())
-    match = SYSTEM_QUERY_RE.match(sql_query)
-    return True if match else False
+    sql_query = sql_query[:-1] if sql_query.endswith(";") else sql_query
```

Comment:
Nit: Consider using sql_query.strip().rstrip(';') to handle this more robustly to  removes any leading or trailing whitespace.